js/src/lib.rs

```diff
@@ -107,6 +107,24 @@ impl JsHandle {
         panic!("unsupported doughnut version");
     }
 
+    /// Sign and return sr25519 signature
+    pub fn sign_sr25519(&mut self, secret_key: &[u8]) -> Vec<u8> {
+        if let Doughnut::V0(mut doughnut) = self.0.clone() {
```

Comment:
- no need to `clone()` here (may also mean that the signature isn't being set as expected)

- the API we're providing is using a builder pattern so would be good to return the doughnut instead of the signature.

- instead of `unwrap()` which will panic, could return a Result type here so on error it will make js exception. To let consumer decide if they panic or not. There are a couple other places we can improve this in future like `fn domain()` and also the "unsupported version" could become an error instead

e.g.
```js
let mySignedDoughnut = new Doughnut(issuer, holder, expiry)
                             .addDomain(..)
                             .sign_sr25519(..);
   
```

Reply:
Was debugging for a while and did not figure out why signature signed but was not able to store in doughnut instance.... Here is the reason, thanks.

Reply:
Umm... Not able to make it happen after a few tries:

Tried to remove `clone()` from `self.0`:
```rust
// Got errors like
// consider borrowing here: `&self.0`;
// move occurs because `doughnut` has type `doughnut_rs::v0::DoughnutV0`, which does not implement the `Copy` trait`;
pub fn sign_sr25519(&mut self, secret_key: &[u8]) -> Vec<u8> {
        if let Doughnut::V0(mut doughnut) = self.0 {
            ...
        }
    }
```

Then I tried with `&self.0`
```rust
// Errors:
// data moved here;
// move occurs because `doughnut` has type `doughnut_rs::v0::DoughnutV0`, which does not implement the `Copy` trait;
pub fn sign_sr25519(&mut self, secret_key: &[u8]) -> Vec<u8> {
        if let Doughnut::V0(mut doughnut) = &self.0 {
            ...
        }
    }
```
Any suggestions? @holygits

Reply:
`&mut self`?